nautobot/tenancy/views.py

```diff
@@ -18,68 +19,67 @@
 #
 
 
-class TenantGroupListView(generic.ObjectListView):
-    queryset = TenantGroup.objects.all()
-    filterset = filters.TenantGroupFilterSet
-    filterset_form = forms.TenantGroupFilterForm
-    table = tables.TenantGroupTable
-
-
-class TenantGroupView(generic.ObjectView):
+class TenantGroupUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.TenantGroupBulkEditForm
+    filterset_class = filters.TenantGroupFilterSet
+    filterset_form_class = forms.TenantGroupFilterForm
+    form_class = forms.TenantGroupForm
     queryset = TenantGroup.objects.all()
+    serializer_class = serializers.TenantGroupSerializer
+    table_class = tables.TenantGroupTable
+    object_detail_content = ObjectDetailContent(
+        panels=(
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+                fields="__all__",
+            ),
+            ObjectsTablePanel(
+                weight=100,
+                section=SectionChoices.RIGHT_HALF,
+                table_class=tables.TenantTable,
+                table_filter="tenant_group",
```

Comment:
Something still seems off here to me - `table_class` and `table_filter` are kwargs to use to auto-create and auto-populate the table. Given you're explicitly providing a table instance in `get_extra_context`, shouldn't this be using `context_table_key="tenant-table"` instead?